# Handling Exceptions 

try: 
  ...
  
except (ValueError, ZeroDivisionError):
  ...
  
else: 
  print("no exceptions raised")
  
finally:
  ...





# Raising exceptions 

if x < 1:   
    raise ValueError(“…”)
 
 
# The with statement 

with open(“file.txt”) as file: 
   ... 
   
   

while True:
    try:
        print("Lastest")            # Executed
        print(1)                    # Executed
        a = int(input("Hel: "))     # Exception will rise if the input is not an integer
        print("Victory\n")
    except:
        print("Defeat \n")






while True:
    try:
        user_input = input("Enter a number between 1 and 10: ")
        number = int(user_input)
        if 1 <= number <= 10:
            break
        else:
            print(f"{number} is not between 1 and 10.")
    except ValueError:
        print(f"{user_input} is not a number.")
